crates/matrix-sdk/src/room/common.rs

```diff
@@ -819,22 +820,49 @@ impl Common {
     /// use matrix_sdk::ruma::events::room::power_levels::SyncRoomPowerLevelsEvent;
     ///
     /// let power_levels: SyncRoomPowerLevelsEvent = room
-    ///     .get_state_event_static("")
+    ///     .get_state_event_static()
     ///     .await?
     ///     .expect("every room has a power_levels event")
     ///     .deserialize()?;
     /// # anyhow::Ok(())
     /// # };
     /// ```
-    pub async fn get_state_event_static<C>(
+    pub async fn get_state_event_static<C>(&self) -> Result<Option<Raw<SyncStateEvent<C>>>>
+    where
+        C: StaticEventContent + StateEventContent<StateKey = EmptyStateKey> + RedactContent,
+        C::Redacted: StateEventContent + RedactedEventContent,
+    {
+        self.get_state_event_static_for_key(&EmptyStateKey).await
+    }
+
+    /// Get a specific state event of statically-known type in this room.
+    ///
+    /// # Example
+    ///
+    /// ```no_run
+    /// # async {
+    /// # let room: matrix_sdk::room::Common = todo!();
+    /// use matrix_sdk::ruma::{
+    ///     events::room::member::SyncRoomMemberEvent, serde::Raw, user_id,
+    /// };
+    ///
+    /// let member_event: Option<Raw<SyncRoomMemberEvent>> = room
+    ///     .get_state_event_static_for_key(user_id!("@alice:example.org"))
+    ///     .await?;
+    /// # anyhow::Ok(())
+    /// # };
+    /// ```
+    pub async fn get_state_event_static_for_key<C, K>(
         &self,
-        state_key: &str,
+        state_key: &K,
```

Comment:
What's the rationale for making this generic instead of keeping `&str`?

Reply:
This way you can pass `&UserId` for a member event, `&RoomId` for a space parent event and such, without `.as_str()`. I actually found a "bug" through this where a test was passing an empty string for getting a member event, which would never return an actual event in the real world.